File: medplum-java/medplum-model/src/main/java/com/fhir/TestScript_Operation.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.Boolean;
import java.lang.Integer;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableTestScript_Operation.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface TestScript_Operation {
  Optional<Coding> type();

  Optional<Boolean> encodeRequestUrl();

  Optional<Testscript_operationMethod> method();

  Optional<String> label();

  Optional<String> params();

  Optional<id> targetId();

  Optional<Integer> origin();

  Optional<String> url();

  Optional<List<Extension>> extension();

  Optional<id> sourceId();

  Optional<id> requestId();

  Optional<code> resource();

  Optional<code> accept();

  Optional<String> description();

  Optional<String> id();

  Optional<code> contentType();

  Optional<List<Extension>> modifierExtension();

  Optional<List<TestScript_RequestHeader>> requestHeader();

  Optional<Integer> destination();

  Optional<id> responseId();

  static ImmutableTestScript_Operation.Builder builder() {
    return ImmutableTestScript_Operation.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/TestScript_Operation.ts
import { code } from "./code";
import { Coding } from "./Coding";
import { Extension } from "./Extension";
import { id } from "./id";
import { Testscript_operationMethod } from "./Testscript_operationMethod";
import { TestScript_RequestHeader } from "./TestScript_RequestHeader";

export interface TestScript_Operation {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    type: Coding | null | undefined;
    resource: code | null | undefined;
    label: string | null | undefined;
    description: string | null | undefined;
    accept: code | null | undefined;
    contentType: code | null | undefined;
    destination: number | null | undefined;
    encodeRequestUrl: boolean | null | undefined;
    method: Testscript_operationMethod | null | undefined;
    origin: number | null | undefined;
    params: string | null | undefined;
    requestHeader: TestScript_RequestHeader[] | null | undefined;
    requestId: id | null | undefined;
    responseId: id | null | undefined;
    sourceId: id | null | undefined;
    targetId: id | null | undefined;
    url: string | null | undefined;
}
